narwhals/_ibis/expr.py

```diff
@@ -0,0 +1,713 @@
+
+    def __eq__(self, other: IbisExpr) -> Self:  # type: ignore[override]
+        return self._with_callable(lambda _input, other: _input == other, other=other)
+
+    def __ne__(self, other: IbisExpr) -> Self:  # type: ignore[override]
+        return self._with_callable(lambda _input, other: _input != other, other=other)
+
+    def __invert__(self) -> Self:
+        invert = cast("Callable[..., ir.Value]", operator.invert)
+        return self._with_callable(invert)
+
+    def alias(self, name: str) -> Self:
+        def alias_output_names(names: Sequence[str]) -> Sequence[str]:
+            if len(names) != 1:
+                msg = f"Expected function with single output, found output names: {names}"
+                raise ValueError(msg)
+            return [name]
+
+        return self.__class__(
+            self._call,
+            evaluate_output_names=self._evaluate_output_names,
+            alias_output_names=alias_output_names,
+            backend_version=self._backend_version,
+            version=self._version,
+        )
+
+    def abs(self) -> Self:
+        return self._with_callable(lambda _input: _input.abs())
+
+    def mean(self) -> Self:
+        return self._with_callable(
+            lambda _input: _input.mean(),
+        )
+
+    def median(self) -> Self:
+        return self._with_callable(lambda _input: _input.median())
+
+    def all(self) -> Self:
+        return self._with_callable(lambda _input: _input.all().name(_input.get_name()))
+
+    def any(self) -> Self:
+        return self._with_callable(lambda _input: _input.any().name(_input.get_name()))
+
+    def quantile(
+        self, quantile: float, interpolation: RollingInterpolationMethod
+    ) -> Self:
+        def func(_input: ir.Column) -> ir.Column:
+            if interpolation == "linear":
+                return cast(
+                    "ir.Column", _input.quantile(quantile).name(_input.get_name())
+                )
+            msg = "Only linear interpolation methods are supported for Ibis quantile."
+            raise NotImplementedError(msg)
+
+        return self._with_callable(func)
+
+    def clip(self, lower_bound: Any, upper_bound: Any) -> Self:
+        def _clip(_input: ir.NumericValue, lower: Any, upper: Any) -> ir.NumericValue:
+            return _input.clip(lower=lower, upper=upper)
+
+        return self._with_callable(_clip, lower=lower_bound, upper=upper_bound)
+
+    def sum(self) -> Self:
+        return self._with_callable(lambda _input: _input.sum())
+
+    def n_unique(self) -> Self:
+        return self._with_callable(
+            lambda _input: _input.nunique() + _input.isnull().any().cast("int8")
+        )
+
+    def count(self) -> Self:
+        return self._with_callable(lambda _input: _input.count())
+
+    def len(self) -> Self:
+        return self._with_callable(
+            lambda _input: _input.count() + _input.isnull().cast("int8").sum()
```

Comment:
does it work to do something like
```python
        def func(df: IbisLazyFrame) -> list[ibis.Expression]:
            return [df.count()]

        return self.__class__(
            func,
            evaluate_output_names=self._evaluate_output_names,
            alias_output_names=self._alias_output_names,
            backend_version=self._backend_version,
            version=self._version,
        )
```
without using `_with_callable`? Looks like Ibis has `table.count` for this, but no expression (🤔 )

Reply:
Ah good idea, works like this with `df.native.count()` 👍